### PR TITLE
nightly running against bucket

### PR DESCRIPTION
**Pull request by guyfreund**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/30824

## Description
- Nightly is now running against the production bucket and not against origin/master when retrieving the latest version of each pack. 
- Refactored the latest pack regex to also catch pack names that consist of a dot.

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [x] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

### CODE REVIEW

**Comment by guyfreund** [comment]
A successful nightly installation: https://app.circleci.com/pipelines/github/demisto/content/46540/workflows/eb4c27a0-d9b0-462f-a2b2-3523adcedd22
See `Install Content And Configure Integrations On Server` step in `Server Master` job.

File: Tests/Marketplace/search_and_install_packs.py [hunk 1]
```diff
@@ -375,41 +379,85 @@ def search_pack_and_its_dependencies(client: demisto_client,
...
+    pack_bucket_path = os.path.join(GCPConfig.STORAGE_BASE_PATH, pack_id)
+    # Adding the '/' in the end of the prefix to search for the exact pack id
+    pack_versions_paths = [f.name for f in production_bucket.list_blobs(prefix=f'{pack_bucket_path}/') if
+                           f.name.endswith('.zip')]
+    pack_versions = [LooseVersion(PACK_PATH_VERSION_REGEX.findall(path)[0]) for path in pack_versions_paths]
+    pack_latest_version = max(pack_versions).vstring
+    return pack_latest_version
+
+
+def add_pack_to_installation_request(pack_id: str, pack_version: str, installation_request_body: List[dict]):
```
Comment (ShahafBenYakir) [inline Tests/Marketplace/search_and_install_packs.py:401]: What is installation_request_body?

File: Tests/scripts/install_content_and_test_integrations.sh [hunk 2]
```diff
@@ -8,9 +8,14 @@ CONF_PATH="./Tests/conf.json"
 
 [ -n "${NIGHTLY}" ] && IS_NIGHTLY=true || IS_NIGHTLY=false
 
+if [ "$IS_NIGHTLY" = true ]; then
+  GCS_PATH=$(mktemp)
+  echo $GCS_MARKET_KEY > $GCS_PATH
+fi
+
```
[the remark this answers is not in the document]
Comment (guyfreund) [reply Tests/scripts/install_content_and_test_integrations.sh:15]: Creating a temp file and writing the `GCS_MARKET_KEY` content (service account JSON) to the temp file.

File: Tests/scripts/install_content_and_test_integrations.sh [hunk 3]
```diff
@@ -8,9 +8,14 @@ CONF_PATH="./Tests/conf.json"
 
 [ -n "${NIGHTLY}" ] && IS_NIGHTLY=true || IS_NIGHTLY=false
 
+if [ "$IS_NIGHTLY" = true ]; then
+  GCS_PATH=$(mktemp)
+  echo $GCS_MARKET_KEY > $GCS_PATH
+fi
```
[the remark this answers is not in the document]
Comment (guyfreund) [reply Tests/scripts/install_content_and_test_integrations.sh:14]: Tried to do this but it didn't work.

File: Tests/Marketplace/search_and_install_packs.py [hunk 4]
```diff
@@ -375,41 +379,85 @@ def search_pack_and_its_dependencies(client: demisto_client,
...
+    all_packs = []
+    logging.debug(f"Installing all content packs in server {host}")
+
+    for pack_id in os.listdir(PACKS_FULL_PATH):
+        if pack_id not in IGNORED_FILES:
+            metadata_path = os.path.join(PACKS_FULL_PATH, pack_id, PACK_METADATA_FILE)
+            with open(metadata_path, 'r') as json_file:
+                pack_metadata = json.load(json_file)
+                pack_version = pack_metadata.get('currentVersion')
+            add_pack_to_installation_request(pack_id, pack_version, all_packs)
```
[the remark this answers is not in the document]
Comment (guyfreund) [reply Tests/Marketplace/search_and_install_packs.py:457]: no, `all_packs` is passed by reference to `add_pack_to_installation_request` and we append there the pack installation data.

File: Tests/Marketplace/search_and_install_packs.py (hunk 1 above)
Comment (guyfreund) [reply Tests/Marketplace/search_and_install_packs.py:401]: it's `all_packs`

File: Tests/configure_and_test_integration_instances.py [hunk 5]
```diff
@@ -157,6 +157,7 @@ def __init__(self, options):
         self.tests_to_run = self.fetch_tests_list(options.tests_to_run)
         self.content_root = options.content_root
         self.pack_ids_to_install = self.fetch_pack_ids_to_install(options.pack_ids_to_install)
+        self.service_account = options.service_account if options.service_account else None
```
[the remark this answers is not in the document]
Comment (guyfreund) [reply Tests/configure_and_test_integration_instances.py:160]: Not sure I understand

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**